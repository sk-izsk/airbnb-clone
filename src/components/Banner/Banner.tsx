import { Box, Button, makeStyles, Typography } from '@material-ui/core';
import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { CustomTheme } from '../../theme/muiTheme';
import { Search } from '../Search/Search';

export interface BannerProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  banner: {
    background: `url("https://media.cntraveler.com/photos/5db1d0dd11c1e500092e7133/master/pass/airbnb-ski-aspen-28328347.jpg") center center`,
    height: '50vh',
    position: 'relative',
  },
  bannerInfo: {
    backgroundColor: 'black',
    color: 'white',
    paddingTop: theme.spacing(6.25),
    paddingLeft: theme.spacing(6.25),
    paddingRight: theme.spacing(6.25),
    paddingBottom: theme.spacing(5),
    width: theme.spacing(43.5),
    height: theme.spacing(54),
  },
  exploreBtn: {
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.secondary.main,
    textTransform: 'inherit',
    marginTop: theme.spacing(1.25),
    fontWeight: 600,
    '&:hover': {
      backgroundColor: theme.palette.secondary.main,
      color: theme.palette.primary.main,
    },
  },
  text: {
    marginTop: theme.spacing(1.25),
  },
  bannerSearch: {
    display: 'flex',
    flexDirection: 'column',
  },
  bannerSearchButton: {
    backgroundColor: theme.palette.secondary.main,
    fontWeight: 900,
    textTransform: 'inherit',
    color: theme.palette.primary.main,
    '&:hover': {
      border: 'none',
    },
  },
}));

const Banner: React.FC<BannerProps> = () => {
  const classes = useStyles();
  const history = useHistory();
  const [showSearch, setShowSearch] = useState<boolean>(false);

  const handleDates = () => {
    setShowSearch(!showSearch);
  };

  return (
    <Box className={classes.banner}>
      <Box className={classes.bannerSearch}>
        {showSearch && <Search />}
        <Button onClick={handleDates} variant='outlined' className={classes.bannerSearchButton}>
          Search Dates
        </Button>
      </Box>
      <Box className={classes.bannerInfo}>
        <Typography variant='h3'>Get out and stretch your imagination</Typography>
        <Typography className={classes.text} variant='h5'>
          Plan a different kind of gateway to uncover the hidden gems near you
        </Typography>
        <Button className={classes.exploreBtn} onClick={() => history.push('/search')} variant='outlined'>
          Explore Nearby
        </Button>
      </Box>
    </Box>
  );
};

export { Banner };
