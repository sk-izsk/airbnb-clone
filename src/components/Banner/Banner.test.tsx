import { render } from '@testing-library/react';
import React from 'react';
import { Banner } from './Banner';

test('Banner renders correctly', () => {
  const { asFragment } = render(<Banner />);
  expect(asFragment()).toMatchSnapshot();
});
