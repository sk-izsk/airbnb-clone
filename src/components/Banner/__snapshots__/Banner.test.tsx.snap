// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-7 makeStyles-banner-1"
  >
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-bannerSearch-5"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-bannerSearchButton-6"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Search Dates
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-bannerInfo-2"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Get out and stretch your imagination
      </h3>
      <h5
        class="MuiTypography-root makeStyles-text-4 MuiTypography-h5"
      >
        Plan a different kind of gateway to uncover the hidden gems near you
      </h5>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-exploreBtn-3"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Explore Nearby
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
